src/enclave_proc/resource_manager.rs

```diff
@@ -905,6 +922,189 @@ impl EnclaveManager {
         Ok(())
     }
 
+    /// Set measurements field inside EnclaveHandle
+    pub fn set_measurements(
+        &mut self,
+        measurements: BTreeMap<String, String>,
+    ) -> NitroCliResult<()> {
+        self.enclave_handle
+            .lock()
+            .map_err(|e| {
+                new_nitro_cli_failure!(
+                    &format!("Failed to acquire lock: {:?}", e),
+                    NitroCliErrorEnum::LockAcquireFailure
+                )
+            })?
+            .measurements = measurements;
+        Ok(())
+    }
+
+    /// Initialize a Sha384 hasher struct
+    pub fn init_sha384_hasher(hasher_name: &str) -> NitroCliResult<EifHasher<Sha384>> {
+        Ok(EifHasher::new_without_cache(Sha384::new()).map_err(|e| {
+            new_nitro_cli_failure!(
+                &format!("Could not create {}: {:?}", hasher_name, e),
+                NitroCliErrorEnum::HasherError
+            )
+        })?)
+    }
+
+    /// Encode hasher
+    pub fn encode_sha384_hasher(
+        mut hasher: EifHasher<Sha384>,
+        hasher_name: &str,
+    ) -> NitroCliResult<String> {
+        Ok(hex::encode(hasher.tpm_extend_finalize_reset().map_err(
+            |e| {
+                new_nitro_cli_failure!(
+                    &format!("Could not get result for {}: {:?}", hasher_name, e),
+                    NitroCliErrorEnum::HasherError
+                )
+            },
+        )?))
+    }
+
+    /// Read EIF, calculate PCR values and save them.
+    ///
+    /// Go through all the section of the EIF file, extract the headers
+    /// and compute the PCRs using the hashers, based on the section type.
+    /// Measurements will be returned as result of a separate thread executing
+    /// this function.
+    pub fn get_pcrs(eif_path: &str) -> NitroCliResult<BTreeMap<String, String>> {
```

Comment:
this function need to be placed in eif_utils, probably in a  struct called EifReader.